linkerd/idle-cache/src/lib.rs

```diff
@@ -329,6 +339,15 @@ impl<V> Cached<V> {
     }
 }
 
+impl<T> Cached<T> {
+    pub fn clone_with<U>(&self, inner: U) -> Cached<U> {
```

Comment:
maybe worth a comment explaining that this creates a new `Cached<U>` that's tied to the eviction lifetime of `self`?